File: php/test/Structural/ProxyTest.php
<?

require_once 'src/Structural/Proxy/PotOfTeaProxy.php';

class Structural_ProxyTest extends PHPUnit_Framework_TestCase
{
   public function test()
   {
		$potOfTea = new PotOfTeaProxy();
		$this->assertEquals( $potOfTea->pourTea() , 'Pouring tea', 'Test pouring from a pot of tea');
	}
}

File: go/structural/Proxy/proxy_test.go
//TestProxy - testing the Proxy









import (
	"testing"
	"../../lib/asserts"
)

func TestProxy {

	@Test
   public void proxy() {
		 
		 PotOfTeaInterface potOfTea = new PotOfTeaProxy()
       asserts.Equals( t, "TestProxy: pouring tea", 
			 "Pouring tea",
			 potOfTea.pourTea()
		)
   }
}
